pkg/sse/client.go

```diff
@@ -0,0 +1,74 @@
+package sse
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+)
+
+// Client represents a sse client
+type Client struct {
+	http http.Client
+}
+
+const (
+	// pipelineSSEUrl is the template for pipeline logs sse event subscription
+	pipelineSSEUrl = "%s/sse/logs/%s/gitdeploy/%s?action=%s"
+)
+
+// NewClient returns a new SSE client
+func NewClient() *Client {
+	return &Client{
+		http: http.Client{},
```

Comment:
Looking at the implementation, seems like more than an `http.Client`, this `Client` depends on a function to get the sse logs. I think it would be better to inject the function, since it will reduce the points of change in this code if some day the sse logs change

It will also make this more testable, and will simplify the function `StreamPipelineLogs`

Reply:
Since this is calling the `oktetoClient` which we already have mocked. Can we move this code to the oktetoClient and make a new function `oktetoClient().Sse().Pipeline()` or `oktetoClient().Pipeline().Sse()`

Reply:
I think what you are referring to is injecting the `io.Reader`. If this client received the io.Reader it would simply transform the response from SSE to the `eventLog`.

The issue I see with a single `Client` is that everything related to SSE will be added to it and we'll end up with a "God class".

We could have on one side the Fetcher/Reader that reads from SSE and then the Transformer which reads from the fetcher and transforms the response (line by line).

This architecture could be later extended to other log resources, events, etc.

Just thinking out loud here, no need to do it now.

Reply:
yup, i've moved the Stream function inside the Pipeline client. here we have to parse the sse event which comes line by line, one line `id`, other `event` and other `data`. we are parsing `data` to be a log object and displaying it by log.Println()